tests/integration/test_types_integration.py

```diff
@@ -200,6 +201,119 @@ def test_digest(trino_connection):
         .execute()
 
 
+def test_date(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DATE)", python=None) \
+        .add_field(sql="DATE '2001-08-22'", python=datetime.date(2001, 8, 22)) \
+        .add_field(sql="DATE '02001-08-22'", python=datetime.date(2001, 8, 22)) \
+        .execute()
+
+
+def test_time(trino_connection):
+    time_0 = datetime.time(1, 23, 45)
+    time_3 = datetime.time(1, 23, 45, 123000)
+    time_6 = datetime.time(1, 23, 45, 123456)
```

Comment:
Can you check whether the JDBC driver truncates or rounds values with higher precision?

Reply:
The Java `Timestamp` type seems to support types up to `TIMESTAMP(9)`. Everything beyond this is truncated, but the nanoseconds are returned by the JDBC driver.

Reply:
Thanks, just wanted to confirm whether behaviour is truncation or rounding. It seems however that the behaviour is rounding by seeing https://github.com/trinodb/trino/blob/a977d88b051e6616d1fd9c355362627383abec29/client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java#L2012.

We should be consistent across drivers where possible.

Reply:
It seems it rounds, but IMO it happens here
https://github.com/trinodb/trino/blob/63f2fdab69094776bc464f2db729f8070ff98505/client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java#L2041-L2044

Reply:
Updated the code to round the milliseconds and updated associated tests